src/rust/engine/logging/src/logger.rs

```diff
@@ -123,7 +146,23 @@ impl Log for Logger {
   fn log(&self, record: &Record) {
     let destination = get_destination();
     match destination {
-      Destination::Stderr => self.stderr_log.lock().log(record),
+      Destination::Stderr => {
+        match self.engine_display_handles.try_lock() {
+          None => (),
+          Some(ref mut map) => {
+            for handle in map.values_mut() {
+              let log_string: String = format!("{}", record.args());
+              match handle.try_lock() {
```

Comment:
The effect of using `try_lock` is I think that if anyone is currently holding the lock, a message will get dropped on the floor. We should try to avoid that if possible.

It looks like the `Scheduler::execute` method currently holds the lock on the `EngineDisplay` for the entire run, bu that is precisely when we're hoping to render messages from logging. So that coarse locking will probably need to change into more fine-grained locking around individual method calls on the Display in `execute`.